### PR TITLE
Fix flakyness of the CapabilityManagementServiceTest due to artifact store setup

### PR DESCRIPTION
**Pull request by pandyajay10**
Also sickbayed a test which is due to known issue.

### CODE REVIEW

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java [hunk 1]
```diff
@@ -169,7 +169,8 @@ public void testCapabilityManagement() throws Exception {
     //cleanup
     new File(externalConfigPath, fileName).delete();
     capabilityManagementService.runTask();
-    artifactRepository.deleteArtifact(Id.Artifact.from(new Id.Namespace(namespace), appName, version));
+    artifactRepository.clear(NamespaceId.SYSTEM);
```
Comment (chtyim) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: Can you move this to a `@After` method to make sure it gets cleanup after every test?

**Review by albertshau — COMMENTED** [review]
normally cleanup is in an `@After` annotated method. For example, if one test fails because something throws an exception before the cleanup, it can cause other tests to fail.

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java [hunk 2]
```diff
@@ -480,12 +481,8 @@ public void testApplicationDeployment() throws Exception {
 
     applicationLifecycleService.removeApplication(NamespaceId.DEFAULT.app(appNameWithCapability, testVersion));
     applicationLifecycleService.removeApplication(NamespaceId.DEFAULT.app(appNameWithOutCapability, testVersion));
-    artifactRepository.deleteArtifact(Id.Artifact
-                                        .from(new Id.Namespace(NamespaceId.DEFAULT.getNamespace()),
-                                              appNameWithCapability, testVersion));
-    artifactRepository.deleteArtifact(Id.Artifact
-                                        .from(new Id.Namespace(NamespaceId.DEFAULT.getNamespace()),
-                                              appNameWithOutCapability, testVersion));
+    artifactRepository.clear(NamespaceId.SYSTEM);
```
Comment (chtyim) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: Still needed?

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java [hunk 3]
```diff
@@ -557,12 +554,10 @@ public void testProgramStart() throws Exception {
     new File(externalConfigPath, capability).delete();
     capabilityManagementService.runTask();
     Assert.assertTrue(capabilityStatusStore.getConfigs(Collections.singleton(capability)).isEmpty());
-    artifactRepository.deleteArtifact(Id.Artifact
-                                        .from(new Id.Namespace(namespace), appName, version));
   }
 
   @Test
-  public void testStatusCheck() throws IOException {
+  public void testStatusCheck() throws Exception {
```
Comment (chtyim) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: I don't think you need any of these signature changes.

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java [hunk 4]
```diff
@@ -624,10 +619,12 @@ public void testGetConfigs() throws IOException {
     //cleanup
     capabilityStatusStore.deleteCapability(testCapability1);
     capabilityStatusStore.deleteCapability(testCapability2);
+    artifactRepository.clear(NamespaceId.SYSTEM);
```
Comment (chtyim) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: Remove?

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java [hunk 5]
```diff
@@ -493,7 +506,7 @@ public void testApplicationDeployment() throws Exception {
     capabilityManagementService.runTask();
   }
 
-  @Test
+  //TODO(CDAP-17528): Re-enable this test once fix.
```
Comment (CuriousVini) [inline cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: why do we need to ignore this test now? 
Wondering if there is any underlying issue with implementation or this is just flaky test? If latter, could we fix it?

Comment (chtyim) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: BTW, use `@Ignore` to ignore test

**Review by chtyim — APPROVED** [review]

File: cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java (hunk 5 above)
Comment (pandyajay10) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: https://cdap.atlassian.net/browse/CDAP-17528 is a known issue hence sickbaying the test. The test was not setup correctly it seems.

Comment (CuriousVini) [reply cdap-app-fabric/src/test/java/io/cdap/cdap/internal/capability/CapabilityManagementServiceTest.java]: +1 to Terence's comment, Also add the correct jira in todo
`CDAP-17589`

**Review by CuriousVini — APPROVED** [review]
Just one minor comment. Please fix, squash and merge.

**Pull request merged**